WordPress/Classes/ViewRelated/Blog/Blogging Reminders/BloggingRemindersPushPromptViewController.swift

```diff
@@ -0,0 +1,248 @@
+        label.textAlignment = .center
+        label.textColor = .secondaryLabel
+        return label
+    }()
+
+    private let hintLabel: UILabel = {
+        let label = UILabel()
+        label.font = .preferredFont(forTextStyle: .body)
+        label.text = TextContent.hint
+        label.numberOfLines = 0
+        label.textAlignment = .center
+        label.textColor = .secondaryLabel
+        return label
+    }()
+
+    private let turnOnNotificationsButton: UIButton = {
+        let button = FancyButton()
+        button.translatesAutoresizingMaskIntoConstraints = false
+        button.isPrimary = true
+        button.setTitle(TextContent.turnOnButtonTitle, for: .normal)
+        button.addTarget(self, action: #selector(turnOnButtonTapped), for: .touchUpInside)
+        return button
+    }()
+
+    private let dismissButton: UIButton = {
+        let button = UIButton(type: .custom)
+        button.translatesAutoresizingMaskIntoConstraints = false
+        button.setImage(.gridicon(.cross), for: .normal)
+        button.tintColor = .secondaryLabel
+        button.addTarget(self, action: #selector(dismissTapped), for: .touchUpInside)
+        return button
+    }()
+
+    // MARK: - Properties
+
+    /// Indicates whether push notifications have been disabled or not.
+    ///
+    private var pushNotificationsAuthorized: UNAuthorizationStatus = .notDetermined {
+        didSet {
+            navigateIfNecessary()
+        }
+    }
+
+    /// Analytics tracker
+    ///
+    let tracker: BloggingRemindersTracker
+
+    // MARK: - Initializers
+
+    init(tracker: BloggingRemindersTracker) {
+        self.tracker = tracker
+
+        super.init(nibName: nil, bundle: nil)
+
+        NotificationCenter.default.addObserver(self, selector: #selector(applicationBecameActive), name: UIApplication.didBecomeActiveNotification, object: nil)
+    }
+
+    required init?(coder: NSCoder) {
+        // This VC is designed to be initialized programmatically.
+        fatalError("Use init(tracker:) instead")
+    }
+
+    // MARK: - View Lifecycle
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        view.backgroundColor = .basicBackground
+        view.addSubview(dismissButton)
+
+        configureStackView()
+
+        view.addSubview(turnOnNotificationsButton)
+        configureConstraints()
+
+        navigationController?.setNavigationBarHidden(true, animated: false)
+    }
+
+    override func viewDidAppear(_ animated: Bool) {
+        tracker.screenShown(.enableNotifications)
+
+        super.viewDidAppear(animated)
+    }
+
+    override func viewDidDisappear(_ animated: Bool) {
+        super.viewDidDisappear(animated)
+
+        // If a parent VC is being dismissed, and this is the last view shown in its navigation controller, we'll assume
+        // the flow was completed.
+        if isBeingDismissedDirectlyOrByAncestor() && navigationController?.viewControllers.last == self {
+            tracker.flowCompleted()
```

Comment:
Assuming there's no weird race-condition scenario where the user manages to schedule the notifications and then dismiss this VC, this should probably always be `tracker.flowDismissed()`.

Reply:
Thanks, good catch!